rocAL/source/readers/file_source_reader.cpp

```diff
@@ -222,58 +273,51 @@ Reader::Status FileSourceReader::generate_file_names() {
     }
 
     if (_file_names.empty())
-        ERR("FileReader ShardID [" + TOSTR(_shard_id) + "] Did not load any file from " + _folder_path)
-
-    // the following code is required to make every shard the same size - required for the multi-gpu training
-    uint images_to_pad_shard = (ceil(_file_count_all_shards / _shard_count) * _shard_count) - _file_count_all_shards;
-    if (!images_to_pad_shard) {
-        for (uint i = 0; i < images_to_pad_shard; i++) {
-            if (get_file_shard_id() != _shard_id) {
-                _file_count_all_shards++;
-                incremenet_file_id();
-                continue;
+        WRN("FileReader ShardID [" + TOSTR(_shard_id) + "] Did not load any file from " + _folder_path)
+
+    auto dataset_size = _file_count_all_shards;
+    // Pad the _file_names with last element of the shard in the vector when _pad_last_batch_repeated is True
+    if (_shard_size > 0)
+        _padded_samples = _shard_size % _batch_count;
+    else
+        _padded_samples = largest_shard_size_without_padding() % _batch_count;
```

Comment:
I think we need to infer the shard_size if it is not set (-1). Instead of using 2 separate variables